systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java

```diff
@@ -203,6 +203,51 @@ void testCruiseControlWithRebalanceResourceAndRefreshAnnotation(ExtensionContext
         KafkaRebalanceUtils.doRebalancingProcess(new Reconciliation("test", KafkaRebalance.RESOURCE_KIND, Constants.TEST_SUITE_NAMESPACE, clusterName), Constants.TEST_SUITE_NAMESPACE, clusterName);
     }
 
+    @IsolatedTest
+    void testCruiseControlSpecChangesReflectedInStatusWhenSpecChanges(ExtensionContext extensionContext) {
```

Comment:
I guess we can maybe put these two tests together or?

Reply:
Iam not sure about that since one test is to check if error is propogated correctly and other is to test if new request completes successfully but yeah I can try to merge them

Reply:
@scholzj @kyguy WDYT? Should I merge these tests or?

Reply:
Not everything needs a system test. So maybe you should cover some of the logic in unit tests.

Reply:
You should not need to use the watch. The watch should just trigger reconciliation, so you can trigger the reconciliation multiple times without the watch. Also, if there is no other way you could use the Mock Kube already used in some KafkaReblanace related tests.

Reply:
I tried as you suggested @scholzj. Just one issue, The refresh annotation is being applied at the moment in the `eventHandler` method. And what I can see is that we are calling `reconcileRebalance` in the KRAO tests during the reconciliation purpose. So the `refresh` annotation is not being set as it is only applied when `eventHandler` method is called which is I guess a part of `createWatch` method. So to make things easier can I apply `refresh` annotation directly and I can comment on it that it is applied as the spec is updated and should be done automatically?

Reply:
Well, as I said, the Mock Kube cluster can work with events as far as I know. So maybe you can use that. Also, if the code is not unit-testable, maybe you should consider some refactoring to make it testable. I'm afraid I do not know the details of how the Kafka Rebalance operator works, so I'm probably not the right one to give you some more detailed instructions. But I do not think running 10 minutes of system tests just because it is not easy to unit test it is the right choice.